plugins/pagerduty/komand_pagerduty/actions/get_on_call/action.py

```diff
@@ -16,34 +16,45 @@ def __init__(self):
         )
 
     def run(self, params={}):
-        schedule_id = params.get(Input.SCHEDULE_ID, None)
-        user_ids = []
-        for oncall_object in self.connection.api.get_on_calls(schedule_id).get("oncalls", []):
-            try:
-                user_ids.append(oncall_object["user"]["id"])
-            except KeyError as e:
-                self.logger.warning(f"User ID not available: {str(e)}")
-                continue
+        schedule_id = params.get(Input.SCHEDULE_ID)
+
+        user_ids = self.get_user_ids(schedule_id=schedule_id)
 
         if not user_ids and schedule_id:
             self.logger.warning(
                 f"No users found for the provided schedule ID - {schedule_id}. "
                 "Please make sure that the schedule used is correct."
             )
 
-        users = asyncio.run(self.async_get_users(user_ids))
-        return {Output.USERS: insightconnect_plugin_runtime.helper.clean(users)}
-
-    async def async_get_users(self, user_ids: [str]) -> list:
-        connection = self.connection.async_connection
-        async with connection.get_async_session() as async_session:
-            tasks: [asyncio.Future] = []
-            for user_id in user_ids:
-                url = f"https://api.pagerduty.com/users/{user_id}"
-                tasks.append(
-                    asyncio.ensure_future(connection.async_request(session=async_session, url=url, method="get"))
-                )
-            user_objects = await asyncio.gather(*tasks)
-            users = [u.get("user") for u in user_objects]
-
-            return users
+        list_of_users = self.get_list_of_users_from_ids(user_ids=user_ids)
+
+        return {Output.USERS: list_of_users}
+
+    def get_user_ids(self, schedule_id: str):
+        user_ids = []
+        for oncall_object in self.connection.api.get_on_calls(schedule_id).get("schedule", {}).get("users", []):
```

Comment:
I would move the api call to a `try: except` statement then loop through the result

Reply:
The error handling for an invalid / bad API call is all handled within the `util/api.py` class  and this will get raised as an plugin except error in there with the error message that `Schedule Not Found` in the case that the schedule id is not valid



```
{
    "body": {
        "error": "An error occurred during plugin execution!\n\nInvalid or unreachable endpoint provided. Verify the URLs or endpoints in your configuration are correct. Response was: {\"error\":{\"code\":3004,\"message\":\"Schedule Not Found\"}}",
        "exception": {
            "assistance": "Verify the URLs or endpoints in your configuration are correct.",
            "cause": "Invalid or unreachable endpoint provided.",
            "data": "{\"error\":{\"code\":3004,\"message\":\"Schedule Not Found\"}}"
        },
        "log": "rapid7/PagerDuty:3.0.0. Step name: get_on_call\nAn error occurred during plugin execution!\n\nInvalid or unreachable endpoint provided. Verify the URLs or endpoints in your configuration are correct. Response was: {\"error\":{\"code\":3004,\"message\":\"Schedule Not Found\"}}\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-5.2.3-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 409, in handle_step\n    output = self.start_step(\n  File \"/usr/local/lib/python3.8/site-packages/insightconnect_plugin_runtime-5.2.3-py3.8.egg/insightconnect_plugin_runtime/plugin.py\", line 604, in start_step\n    output = func(params)\n  File \"/usr/local/lib/python3.8/site-packages/pagerduty_rapid7_plugin-3.0.0-py3.8.egg/komand_pagerduty/actions/get_on_call/action.py\", line 21, in run\n    user_ids = self.get_user_ids(schedule_id=schedule_id)\n  File \"/usr/local/lib/python3.8/site-packages/pagerduty_rapid7_plugin-3.0.0-py3.8.egg/komand_pagerduty/actions/get_on_call/action.py\", line 35, in get_user_ids\n    for oncall_object in self.connection.api.get_on_calls(schedule_id).get(\"schedule\", {}).get(\"users\", []):\n  File \"/usr/local/lib/python3.8/site-packages/pagerduty_rapid7_plugin-3.0.0-py3.8.egg/komand_pagerduty/util/api.py\", line 30, in get_on_calls\n    return self.send_request(\"GET\", f\"/schedules/{schedule_id}/\")\n  File \"/usr/local/lib/python3.8/site-packages/pagerduty_rapid7_plugin-3.0.0-py3.8.egg/komand_pagerduty/util/api.py\", line 207, in send_request\n    self._check_status_code(response)\n  File \"/usr/local/lib/python3.8/site-packages/pagerduty_rapid7_plugin-3.0.0-py3.8.egg/komand_pagerduty/util/api.py\", line 225, in _check_status_code\n    raise PluginException(preset=PluginException.Preset.NOT_FOUND, data=response.text)\ninsightconnect_plugin_runtime.exceptions.PluginException: An error occurred during plugin execution!\n\nInvalid or unreachable endpoint provided. Verify the URLs or endpoints in your configuration are correct. Response was: {\"error\":{\"code\":3004,\"message\":\"Schedule Not Found\"}}\n",
        "meta": {},
        "status": "error"
    },
    "type": "action_event",
    "version": "v1"
}
```